benchmarks/bench_gap_divergence.py

```diff
@@ -0,0 +1,201 @@
+                    rescale_W=self.rescale_W,
+                    gamma_shape_prior=self.gamma_shape_prior,
+                    gamma_scale_prior=self.gamma_scale_prior,
+                )
+                # Update the topics self.W_
+                _multiplicative_update_w(
+                    unq_V[idx],
+                    self.W_,
+                    self.A_,
+                    self.B_,
+                    unq_H[idx],
+                    self.rescale_W,
+                    self.rho_,
+                )
+
+                if i == n_batch - 1:
+                    # Compute the norm of the update of W in the last batch
+                    W_change = np.linalg.norm(self.W_ - W_last) / np.linalg.norm(W_last)
+
+            print(self.W_.shape, self.A_.shape, self.B_.shape)
+
+            self.benchmark_results_.append(
+                {
+                    "score": score,
+                    "W_change": W_change,
+                    "W_": self.W_.copy(),
+                    "A_": self.A_.copy(),
+                    "B_": self.B_.copy(),
+                }
+            )
+
+            if (W_change < self.tol) and (n_iter_ >= self.min_iter - 1):
+                break  # Stop if the change in W is smaller than the tolerance
+
+        # Update self.H_dict_ with the learned encoded vectors (activations)
+        self.H_dict_.update(zip(unq_X, unq_H))
+        return self
+
+
+class ModifiedGapEncoder(GapEncoder):
+    fitted_models_: List[ModifiedGapEncoderColumn]
+
+    @property
+    def benchmark_results_(self):
+        return self.fitted_models_[0].benchmark_results_
+
+    def _create_column_gap_encoder(self):
+        return ModifiedGapEncoderColumn(
+            ngram_range=self.ngram_range,
+            n_components=self.n_components,
+            analyzer=self.analyzer,
+            gamma_shape_prior=self.gamma_shape_prior,
+            gamma_scale_prior=self.gamma_scale_prior,
+            rho=self.rho,
+            rescale_rho=self.rescale_rho,
+            batch_size=self.batch_size,
+            tol=self.tol,
+            hashing=self.hashing,
+            hashing_n_features=self.hashing_n_features,
+            max_iter=self.max_iter,
+            init=self.init,
+            add_words=self.add_words,
+            random_state=self.random_state,
+            rescale_W=self.rescale_W,
+            max_iter_e_step=self.max_iter_e_step,
+        )
+
+
+benchmark_name = "gap_divergence"
+
+
+@monitor(
+    parametrize={
+        "max_iter_e_step": range(1, 21),
+    },
+    save_as=benchmark_name,
+)
+def benchmark(max_iter_e_step: int):
+    """
+    Fit a modified `GapEncoder` instance to a single high cardinality column.
+    """
+    gap = ModifiedGapEncoder(
+        min_iter=5,
+        max_iter=5,
+        max_iter_e_step=max_iter_e_step,
+        random_state=0,
+    )
+    gap.fit(fetch_employee_salaries().X[["employee_position_title"]])
+
+    results: List[Dict[str, Union[float, ...]]] = []
```

Comment:
Nitpick: while I understand the need to type stuff in public function arguments, I don't think we need it for our in-house benchmarks, especially for local variables. This also adds complexity to the reader and the programmer as well. I'd advocate for staying lean in our benchmarks to get results quickly without over-engineering things.

Reply:
I didn't say the other one was not useful ;)
I'm not familiar enough with typing in workflow though; how does it help you avoid the error you mention?